src/App.jsx

```diff
@@ -14,45 +16,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alpha';
+const SORT_BY_LENGTH = 'length';
+
+function sortBy(fieldForSort, isReversed) {
+  const goods = [...goodsFromServer];
+
+  if (fieldForSort) {
+    switch (fieldForSort) {
+      case SORT_BY_ALPHABET:
+        goods.sort((a, b) => a.localeCompare(b));
+        break;
+
+      case SORT_BY_LENGTH:
+        goods.sort((a, b) => a.length - b.length);
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  return isReversed ? goods.reverse() : goods;
+}
+
+export const App = () => {
+  const [isReverseActive, setIsReverseActive] = useState(false);
+  const [sortField, setSortField] = useState('');
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const preparedGoods = sortBy(sortField, isReverseActive);
+
+  const resetAllSort = () => {
+    setIsReverseActive(false);
+    setSortField('');
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => {
+            setSortField(SORT_BY_ALPHABET);
+            sortBy(SORT_BY_ALPHABET, isReverseActive);
+          }}
+          type="button"
+          className={
+          classnames('button',
+            'is-info',
+            { 'is-light': sortField !== SORT_BY_ALPHABET })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => {
+            setSortField(SORT_BY_LENGTH);
+            sortBy(SORT_BY_LENGTH, isReverseActive);
+          }}
+          type="button"
+          className={
+          classnames('button',
+            'is-success',
+            { 'is-light': sortField !== SORT_BY_LENGTH })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => {
+            setIsReverseActive(!isReverseActive);
+          }}
+          type="button"
+          className={
+          classnames(
+            'button',
+            'is-warning',
+            { 'is-light': !isReverseActive },
+          )}
+        >
+          Reverse
+        </button>
+
+        {(sortField || isReverseActive) && (
```

Comment:
Better to create a variable flag for `sortField || isReverseActive` so you can avoid to combine `||` and`&&` operators